Userland/Libraries/LibCore/Group.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/Group.h>
+#include <LibCore/System.h>
+
+namespace Core {
+
+#ifndef AK_OS_MACOS
+ErrorOr<void> Group::add_group(Group& group)
+{
+    if (group.name().is_empty())
+        return Error::from_string_literal("Group name can not be empty.");
+
+    // A quick sanity check on group name
+    if (strpbrk(group.name().characters(), "\\/!@#$%^&*()~+=`:\n"))
+        return Error::from_string_literal("Group name has invalid characters.");
+
+    // Disallow names starting with '_', '-' or other non-alpha characters.
+    if (group.name().starts_with('_') || group.name().starts_with('-') || !is_ascii_alpha(group.name().characters()[0]))
+        return Error::from_string_literal("Group name has invalid characters.");
+
+    // Verify group name does not already exist
+    if (TRY(name_exists(group.name())))
+        return Error::from_string_literal("Group name already exists.");
+
+    // Sort out the group id for the group
+    if (group.id() > 0) {
+        if (TRY(id_exists(group.id())))
+            return Error::from_string_literal("Group ID already exists.");
+    } else {
+        unsigned group_id = 100;
+        while (true) {
+            if (TRY(id_exists(group_id)))
+                break;
+            group_id++;
+        }
+        group.set_group_id(group_id);
+    }
+
+    auto gr = TRY(group.to_libc_group());
+
+    errno = 0;
+    FILE* file = fopen("/etc/group", "a");
+    if (errno)
+        return Error::from_errno(errno);
+
+    int rc = putgrent(&gr, file);
+    if (rc < 0)
```

Comment:
Since we don't need `rc` for anything after this point, I suggest losing the local:
```cpp
if (putgrent(&gr, file) < 0)
```